private/buf/cmd/buf/command/config/configlsmodules/configlsmodules.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/spf13/pflag"
+)
+
+const (
+	configFlagName = "config"
+	formatFlagName = "format"
+
+	formatPath = "path"
+	formatName = "name"
+	formatJSON = "json"
+
+	defaultFormat = formatPath
+)
+
+var (
+	allFormats = []string{
+		formatPath,
+		formatName,
+		formatJSON,
+	}
+)
+
+// NewCommand returns a new Command.
+func NewCommand(
+	name string,
+	builder appext.SubCommandBuilder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name,
+		Short: "List configured modules",
+		Args:  appcmd.NoArgs,
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appext.Container) error {
+				return run(ctx, container, flags)
+			},
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Config string
+	Format string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Config,
+		configFlagName,
+		"",
+		`The buf.yaml file or data to use for configuration.`,
+	)
+	flagSet.StringVar(
+		&f.Format,
+		formatFlagName,
+		defaultFormat,
+		fmt.Sprintf(
+			"The format to print rules as. Must be one of %s",
+			stringutil.SliceToString(allFormats),
+		),
+	)
+}
+
+func run(
+	ctx context.Context,
+	container appext.Container,
+	flags *flags,
+) error {
+	externalModules, err := getExternalModules(ctx, flags.Config)
+	if err != nil {
+		return err
+	}
+	return printExternalModules(ctx, container, flags.Format, externalModules)
+}
+
+func getExternalModules(
+	ctx context.Context,
+	configOverride string,
+) ([]*externalModule, error) {
+	// First, look for a buf.work.yaml file.
+	bufWorkYAMLFile, err := bufcli.GetBufWorkYAMLFileForDirPath(ctx, ".")
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			return nil, err
+		}
+		// We do not have a buf.work.yaml file, attempt to read a buf.yaml file.
+		bufYAMLFile, err := bufcli.GetBufYAMLFileForDirPathOrOverride(ctx, ".", configOverride)
+		if err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				return nil, err
+			}
+			// We do not have a buf.work.yaml or buf.yaml file, use the default.
+			bufYAMLFile, err = bufconfig.NewBufYAMLFile(
+				bufconfig.FileVersionV2,
+				[]bufconfig.ModuleConfig{
+					bufconfig.DefaultModuleConfigV2,
+				},
+				nil,
+			)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// This handles both buf.yaml file and no file courtesy of the above logic.
+		return getExternalModulesForBufYAMLFile(ctx, bufYAMLFile)
+	}
+	// We did have a buf.work.yaml file, but before handling it, check there is not a buf.yaml.
+	_, err = bufcli.GetBufYAMLFileForDirPath(ctx, ".")
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return nil, err
+	}
+	if err == nil {
+		return nil, errors.New("Both buf.work.yaml and buf.yaml found. It is not valid to have a buf.work.yaml and buf.yaml in the same directory, buf.work.yaml specifies a workspace of modules, while buf.yaml either specifies a single module or a workspace of modules itself.")
```

Comment:
Do we want a test for this condition?

Reply:
Yes, it's tested in [TestLsModulesBothConfig](https://github.com/bufbuild/buf/pull/3081/files#diff-ba50276a547effd5a480b68e54d606c6772f6bc6a85082f745aef23955486827R1436)